src/node.cc

```diff
@@ -1100,18 +1100,76 @@ int Start(int argc, char** argv) {
     return result.exit_code;
   }
 
-  {
+  if (!per_process::cli_options->snapshot_main.empty()) {
+    SnapshotData data;
+    {
+      std::string entry;
+      int r =
+          ReadFileSync(&entry, per_process::cli_options->snapshot_main.c_str());
+      if (r != 0) {
+        const char* code = uv_err_name(r);
+        const char* message = uv_strerror(r);
+        FPrintF(stderr,
+                "Failed to open %s. %s: %s\n",
+                per_process::cli_options->snapshot_main.c_str(),
+                code,
+                message);
+        return 1;
+      }
+      node::SnapshotBuilder::Generate(
+          &data, entry, result.args, result.exec_args);
+    }
+
+    std::string snapshot_blob_path;
+    if (!per_process::cli_options->snapshot_blob.empty()) {
+      snapshot_blob_path = per_process::cli_options->snapshot_blob;
+    } else {
+      snapshot_blob_path = std::string("snapshot.blob");
+      char buf[PATH_MAX_BYTES];
+      size_t cwd_size = sizeof(buf);
+      if (uv_cwd(buf, &cwd_size)) {
+        snapshot_blob_path =
+            std::string(buf) + kPathSeparator + std::string("snapshot.blob");
```

Comment:
This doesn’t seem quite right, `uv_cwd(...)` returns an error code and 0 if it succeeded.

But I’m not sure why you’d need to query `uv_cwd()` in the first place – you can just always read from `snapshot.blob`, with the advantage that that also works if the current working directory has been deleted.

Reply:
Right, I forgot to put `== 0` at the end and didn't hit this when I tested it locally, thanks for catching it :) Also, yeah, we probably don't need the cwd call anyway..